part03/chap26/chap26.md

```diff
@@ -0,0 +1,671 @@
+| `? :`  |  10  |  右  | 3項節中   | 条件                     |
+| `\|\|` |   9  |  左  | 2項節中   | 論理和                   |
+| `&&`   |   8  |  左  | 2項節中   | 論理積                   |
+| `==`   |   7  |  左  | 2項節中   | 等しい                   |
+| `!=`   |   7  |  左  | 2項節中   | 等しくない               |
+| `<`    |   6  |  左  | 2項節中   | より小さい               |
+| `<=`   |   6  |  左  | 2項節中   | より小さいか等しい       |
+| `>`    |   6  |  左  | 2項節中   | より大きい               |
+| `>=`   |   6  |  左  | 2項節中   | より大きいか等しい       |
+| `+`    |   5  |  左  | 2項節中   | 加算                     |
+| `-`    |   5  |  左  | 2項節中   | 減算                     |
+| `*`    |   4  |  左  | 2項節中   | 乗算                     |
+| `/`    |   4  |  左  | 2項節中   | （右）除算               |
+| `%`    |   4  |  左  | 2項節中   | 剰余                     |
+| `\\`   |   3  |  左  | 2項節中   | 左除算                   |
+| `.*`   |   2  |  左  | 2項節中   | 要素ごとの乗算           |
+| `./`   |   2  |  左  | 2項節中   | 要素ごとの除算           |
+| `!`    |   1  |  n/a | 単項接頭  | 論理否定                 |
+| `-`    |   1  |  n/a | 単項接頭  | 負                       |
+| `+`    |   1  |  n/a | 単項接頭  | 正（Stanでは無演算）     |
+| `^`    |  0.5 |  右  | 2項節中   | 累乗                     |
+| `'`    |   0  |  n/a | 単項接尾  | 転値                     |
+| `()`   |   0  |  n/a | 接頭,前後 | 関数適用                 |
+| `[]`   |   0  |  左  | 接頭,前後 | 配列・行列インデキシング |
+
+図3.1: Stanの単項・2項・3項演算子における、優先順位、結合性、式中の位置と説明。最後の2行は関数適用と、配列・行列・ベクトルのインデキシングです。演算子は、優先順位にしたがって載せており、上の方がより弱く、下の方がより強く結合します。正しい引数と、対応する戻り値の組み合わせ全体は、第7部の関数の説明のうち、`operator`が前にあるものを見てください（`operator*(int, int):int`は、乗算演算子を2つの整数に適用し、整数を返すことを示しています）。括弧は、優先順位と結合性に依存せず、明示的に式をグルーピングするのにも使われます。
+
+### 3.6. 条件演算子
+
+#### 条件演算子の文法
+
+3項条件演算子は、3つの引数をとるところと、混合文法（**??mixed syntaxの訳語**）をつかうというところで独特です。`a`が`int`型の式で、`b`と`c`が互いに変換できる（すなわち`==`で比較できる）式のとき、
+
+```
+a ? b : c
+```
+
+上の式は`b`と`c`とで昇格した型の式になります。Stanで認められている昇格は、整数から実数のみです。一方が`int`型で他方が`real`型のとき、条件式は全体として`real`型になります。その他の場合はすべて、これら引数は同じStanの基本型である必要があり（制約は関係なく、形式だけです）、条件式はその型となります。
+
+##### 条件演算子の優先順位
+
+条件演算子はもっとも優先順位が低い演算子です。そのため、あいまいさを避けるために引数に括弧を使う必要があることは稀です。例えば、
+
+```
+a > 0 || b < 0 ? c + d : e - f
+```
+
+は、明示的にグルーピングした以下の式と等価です。
+
+```
+(a > 0 || b < 0) ? (c + d) : (e - f)
+```
+
+括弧は厳密には必要ないとしても、後者の方が読みやすいでしょう。
+
+##### 条件演算子の結合性
+
+条件演算子は右から結合します。つまり、
+
+```
+a ? b : c ? d : e
+```
+
+は、明示的にグルーピングすると以下のようになります。
+
+```
+a ? b : (c ? d : e)
+```
+
+前と同じく、明示的にグルーピングした方が読みやすいでしょう。
+
+#### 条件演算子の意味論
+
+Stanの条件演算子は、C++にあるものとほとんど同様に動作します。最初の引数は、整数となる式でなくてはなりません。普通ここは、上の例の`a`のように変数または関係演算子です。次に2つの結果の引数があり、1番目は条件が真（すなわち非零）の場合、2番目は条件が偽（すなわち零）の場合のものです。上の例では、条件が非零値と評価されれば`b`の値が返され、条件が例と評価されれば`c`が返されます。
+
+##### 結果の遅延評価
+
+なぜ高性能コンピューティング（**high performance computingの訳**）で条件演算子がとても有用かというと、戻り値となる部分式のみが評価され、そうでない式は評価されないという特性が鍵です。言い換えると、通常の関数とは違って、関数に値を渡すために引数の式を残らず（**eagerlyの意訳**）評価するということをしません。とくに、組込みの`if_else`関数よりも条件演算子の方がかなり高速になります。`if_else`関数はほかの関数と同様、呼び出されたときに3つの引数をすべて評価するのです。ほかの例と同じく、この節約のほとんどは、基本的な関数評価自身というより、導関数を計算しないことによるものです。
```

Comment:
ほかの例と同じく、この節約のほとんどは、基本的な関数評価自身というより、導関数を計算しないことによるものです。
↓
ほかの例と同じく、この節約のほとんどは、基本的な関数評価そのものではなく、導関数を計算しないことによります。

※若干わかりやすいか、くらいの意図でrather thanの訳を少し変えてみました。